### ISSUE TITLE
Further effects json support (morale, focus)

### ISSUE THREAD

**Issue by I-am-Erk** [CleverRaven/Cataclysm-DDA#35882]
**Is your feature request related to a problem? Please describe.**  
I was creating an effect that impacts morale over time, and realized it is not possible to define effects that impact morale or focus in JSON. Or, if it is, it is not documented.

**Describe the solution you'd like**  
A json effect flag for "morale_mod" similar to every other effect flag. I don't think morale effects need to be impacted by all the other effect stuff like _min / _max / _tick.

Stretch goal: While in there it might be reasonable to have effects that directly impact Focus as well, since this is a stat that should be impacted by a lot of drugs and things. That would probably need to be a full set of effect flags, focus_amount and focus_min and such.

**Describe alternatives you've considered**  
Hacking together effects in hard code, but no.

**Additional context**  
I need this to make it depressing to eat gross protein bars all the time and I can't :(

I believe this is a fairly simple C++ addition and might be easy for someone trying to learn to JSONify C++ code... I think. Take that with a grain of salt since I cannot in fact code in C++

**Comment by DaviBones** [comment]
> I need this to make it depressing to eat gross protein bars all the time and I can't :(

Errr... way ahead of you? I think? #35813

**Comment by I-am-Erk** [comment]
Yeah, similar need for better granularity but in this case more of a "omg I need to stop living off these terrible protein bars" thing.

**Comment by I-am-Erk** [comment]
We're still going to need this for jsonizing drugs and things

**Comment by scrotty** [comment]
I am going to take a stab at this. I am relatively new to the game and very new to its code, but I'm motivated.

**Comment by I-am-Erk** [comment]
Cool, tag me on the pr

**Comment by scrotty** [comment]
@I-am-Erk: I've been looking through the code today and see that the general pattern is to move move control into the json files. But there is still a lot of game behavior in the code. 

Given that, I'd want to make sure I implement the functionality you want. Above you state you want "A json effect flag for "morale_mod" similar to every other effect flag". Can you please give me an example of one of these other effect flags you want morale to model? Or put another way, what would you expect the json to look like one the feature is in place.

For example, here is a portion of the json for aspirin:
```
   "use_action": {
      "type": "consume_drug",
      "activation_message": "You take some aspirin.",
      "effects": [ { "id": "pkill1", "duration": 720 } ]
    }

Which references this effect
  {
    "type": "effect_type",
    "id": "pkill1",
    "max_duration": "4 m",
    "base_mods": { "pkill_tick": [ 42 ], "pkill_min": [ 1 ], "pkill_max_val": [ 15 ] }
  },

```
Would you want something like?:

```
   "use_action": {
      "type": "consume",
      "activation_message": "Ugh. The protein bar sits in your stomach like a brick.",
      "effects": [ 
           { "id": "moraledebuff10", "duration": 720 },
       ]
    }

To reference a new effect:
  {
    "type": "effect_type",
    "id": "moraledebuff10",
    "max_duration": "4 m",
    "base_mods": { "moraledebuff10_tick": [ 42 ], "moraledebuff10_min": [ 1 ], "moraledebuff10_max_val": [ 10 ] }
  },
```

**Comment by Maleclypse** [comment]
Do @Ramza13 's `effect_on_condition` PR's solve this or at least begin work on it?

**Comment by actual-nh** [comment]
In #48683, Ramza13 stated (admittedly a bit back) that one for morale was waiting on a use case.

**Comment by Ramza13** [comment]
Yup I have that already written.  I'll make a PR.

**Comment by I-am-Erk** [comment]
Awesome. Pester me when it's done and I'll see about adjusting some drugs.

**Comment by Ramza13** [comment]
@I-am-Erk There should now be the tools you need to do this in json.  Let me know if you need anything else.

**Comment by Maleclypse** [comment]
@I-am-Erk Do we need to close this issue?
